shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -296,6 +300,16 @@ private void removeView(Object[] args) {
             mFakeWindowRootView.removeView(view);
         }
 
+        private void removeViewImmediate(Object[] args) {
+            if (mFakeWindowRootView == null) {
+                Log.w(TAG, "Embedded view called removeViewImmediate while detached from presentation");
+                return;
+            }
+            View view = (View) args[0];
+            view.clearAnimation();
```

Comment:
Do you know if clearAnimation is triggered when calling WindowManager.removeViewImmediate, and WindowManager.removeView we should probably be consistent with whatever WindowManager is doing.

Doesn't have to happen in this PR, feel free to just leave a:
`// TODO(amirh): Make sure that the clearAnimation call in removeViewImmediate and removeView is consistent with WindowManager.`
And I'll take a look when I'm bacl.

Reply:
Sorry for the late reply.

As far as I know, `WindowManager.removeViewImmediate` will not trigger `View.clearAnimation`, but `mCurrentAnimation` of the view will be assigned to null when it's detached from window.

Clear animation is a workaround for remove view **immediately**, otherwise the view will not got removed synchronously.